computation.cpp

```diff
@@ -55,26 +56,127 @@ std::unique_ptr<ComputationResult> Computation::compute_raw(ComputationInput& in
         }
     }
 
-    //STAGE 3: COMPUTE MULTIGRADED BETTI NUMBERS
+    //STAGE 3: COMPUTE MINIMAL PRESENTTION AND MULTIGRADED BETTI NUMBERS
 
     std::unique_ptr<ComputationResult> result(new ComputationResult);
-    //compute xi_0 and xi_1 at all bigrades
-    if (verbosity >= 2) {
-        debug() << "COMPUTING xi_0, xi_1, AND xi_2 FOR HOMOLOGY DIMENSION " << params.dim << ":";
-    }
-    MultiBetti mb(input.bifiltration(), params.dim);
-    Timer timer;
-    mb.compute(result->homology_dimensions, progress);
-    mb.compute_xi2(result->homology_dimensions);
 
+    MultiBetti mb(input.rep());
+
+    Timer timer;
+    Timer timer_sub;
+    timer.restart();
+    timer_sub.restart();
+    
+    Presentation pres;
+    
+    // If the --koszul flag is not given, then we compute Betti numbers by
+    //computing a minimal presentation
+    if (!koszul)
+    {
```

Comment:
This is enough code that it might make sense to extract a separate method for it.